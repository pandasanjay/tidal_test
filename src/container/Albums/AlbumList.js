import React, { useState } from "react";
const Album = ({ list = [], onSelectAlbum = () => {} }) => {
  const [active, setActive] = useState("");
  const handelClick = id => {
    setActive(id);
    onSelectAlbum(id);
  };
  return (
    <>
      <div className="flex-container scroll">
        {list.map((data, i) => (
          <div
            key={i}
            className={
              "flex text-center album-section " +
              (active === data.id ? "active" : "")
            }
            onClick={() => handelClick(data.id)}
          >
            <img src={data.cover} alt={data.title} />
            <div className="title-text-sm">{data.title}</div>
          </div>
        ))}
      </div>
    </>
  );
};
export default Album;
