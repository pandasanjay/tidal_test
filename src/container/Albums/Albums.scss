@import "../../config";
.search-result-album-section {
  padding: $padding;
  h4 {
    padding: $padding 0;
    border-bottom: 1px solid $separator-color;
    font-weight: 400;
  }
}
.scroll {
  flex-wrap: nowrap;
  overflow-x: scroll;
  overflow-y: hidden;
  -webkit-overflow-scrolling: touch;
  -ms-overflow-style: -ms-autohiding-scrollbar;
}
.scroll::-webkit-scrollbar {
  display: none;
}
.album-section {
  width: 125px;
  cursor: pointer;
  &:nth-child(1) {
    margin-left: 0px;
  }
  &:active,
  &.active {
    border-bottom: 1px solid $separator-color;
  }
}
