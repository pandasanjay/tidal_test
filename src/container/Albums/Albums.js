import React from "react";
import { connect } from "react-redux";

import AlbumList from "../../components/AlbumList/AlbumList";
import AlbumDetails from "../../components/AlbumDetails/AlbumDetails";

import "./Albums.scss";

import mapDispatchToProps from "../../redux/actions";
const mapStateToProps = state => ({
  ...state
});

const Albums = ({ app, findAlbumDetailsByArtist }) => {
  return app.albums ? (
    <>
      <div className="search-result-album-section">
        <h4>Search Results for "{app.searchText}"</h4>
        <div className="title-text pad-t-10 pad-b-10">ALBUMS</div>
        <AlbumList list={app.albums} onChange={findAlbumDetailsByArtist} />
        <AlbumDetails details={app.albumsDetails} />
      </div>
    </>
  ) : (
    <></>
  );
};
export default connect(mapStateToProps, mapDispatchToProps)(Albums);
