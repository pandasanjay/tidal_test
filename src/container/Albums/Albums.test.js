import React from "react";
import configureStore from "redux-mock-store";
import { Provider } from "react-redux";
import { render } from "@testing-library/react";
import Albums from "./Albums";

const mockStore = configureStore([]);
const store = mockStore({
  app: {
    searchText: "Foo",
    albums: [],
    findAlbumDetailsByArtist: jest.fn(),
    albumsDetails: {}
  }
});
jest.mock("../../components/AlbumList/AlbumList", () => "div");
jest.mock("../../components/AlbumDetails/AlbumDetails", () => "div");

it("renders without crashing", () => {
  const { asFragment } = render(
    <Provider store={store}>
      <Albums />
    </Provider>
  );
  expect(asFragment()).toMatchSnapshot();
});
