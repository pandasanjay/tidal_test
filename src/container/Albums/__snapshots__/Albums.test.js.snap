// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<DocumentFragment>
  <div
    class="search-result-album-section"
  >
    <h4>
      Search Results for "Foo"
    </h4>
    <div
      class="title-text pad-t-10 pad-b-10"
    >
      ALBUMS
    </div>
    <div
      list=""
    />
    <div
      details="[object Object]"
    />
  </div>
</DocumentFragment>
`;
